ext/opentelemetry-ext-docker-tests/tests/mysql/test_mysql_functional.py

```diff
@@ -36,35 +37,42 @@
 class TestFunctionalMysql(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
+        cls._connection = None
+        cls._cursor = None
         cls._tracer_provider = TracerProvider()
         cls._tracer = Tracer(cls._tracer_provider, None)
         cls._span_exporter = InMemorySpanExporter()
         cls._span_processor = SimpleExportSpanProcessor(cls._span_exporter)
         cls._tracer_provider.add_span_processor(cls._span_processor)
         trace_integration(cls._tracer)
-        # Try to connect to DB
-        for i in range(5):
-            try:
-                cls._connection = mysql.connector.connect(
-                    user=MYSQL_USER,
-                    password=MYSQL_PASSWORD,
-                    host=MYSQL_HOST,
-                    port=MYSQL_PORT,
-                    database=MYSQL_DB_NAME,
-                )
-                cls._cursor = cls._connection.cursor()
-                break
-            except mysql.connector.Error as err:
-                print(err)
-            time.sleep(5)
 
     @classmethod
     def tearDownClass(cls):
-        cls._connection.close()
+        if cls._connection:
+            cls._connection.close()
 
     def setUp(self):
         self._span_exporter.clear()
 
+    @fixture(autouse=True)
+    def connect_to_db(self):
+        if self._connection is None:
+            # Try to connect to DB
+            for i in range(5):
```

Comment:
This `for` needs an `else` clause. If after 5 attempts no connection has been achieved, then an exception needs to be raised in order for Pytest to avoid running the test cases where this is setup.

Reply:
I've left a similar comment in the postgres tests. I'm voting for fewer retries (if we actually need them), and just re-raising the exception on the last try.